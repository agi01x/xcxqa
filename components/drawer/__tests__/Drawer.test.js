import React from 'react';
import { render } from 'enzyme';
import Drawer from '..';

describe('Drawer', () => {
  it('render correctly', () => {
    const wrapper = render(
      <Drawer
        visible
        width={400}
        getContainer={false}
      >
        Here is content of Drawer
      </Drawer>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render top drawer', () => {
    const wrapper = render(
      <Drawer
        visible
        height={400}
        placement="top"
        getContainer={false}
      >
        Here is content of Drawer
      </Drawer>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('have a title', () => {
    const wrapper = render(
      <Drawer
        visible
        title="Test Title"
        getContainer={false}
      >
        Here is content of Drawer
      </Drawer>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('closable is false', () => {
    const wrapper = render(
      <Drawer
        visible
        closable={false}
        getContainer={false}
      >
        Here is content of Drawer
      </Drawer>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('destroyOnClose is true', () => {
    const wrapper = render(
      <Drawer
        destroyOnClose
        visible={false}
        getContainer={false}
      >
        Here is content of Drawer
      </Drawer>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('className is test_drawer', () => {
    const wrapper = render(
      <Drawer
        destroyOnClose
        visible={false}
        className="test_drawer"
        getContainer={false}
      >
        Here is content of Drawer
      </Drawer>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
