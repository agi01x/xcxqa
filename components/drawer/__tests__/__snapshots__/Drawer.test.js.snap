// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer className is test_drawer 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right test_drawer"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
      style="transform:translateX(100%);-ms-transform:translateX(100%);width:256px"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          class="ant-drawer-wrapper-body"
          style="overflow:auto;height:100%;opacity:0;transition:opacity .3s"
        >
          <button
            aria-label="Close"
            class="ant-drawer-close"
          >
            <span
              class="ant-drawer-close-x"
            >
              <i
                class="anticon anticon-close"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="close"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </i>
            </span>
          </button>
          <div
            class="ant-drawer-body"
          >
            Here is content of Drawer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer closable is false 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
      style="transform:translateX(100%);-ms-transform:translateX(100%);width:256px"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          class="ant-drawer-wrapper-body"
          style="overflow:auto;height:100%"
        >
          <div
            class="ant-drawer-body"
          >
            Here is content of Drawer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer destroyOnClose is true 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
      style="transform:translateX(100%);-ms-transform:translateX(100%);width:256px"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          class="ant-drawer-wrapper-body"
          style="overflow:auto;height:100%;opacity:0;transition:opacity .3s"
        >
          <button
            aria-label="Close"
            class="ant-drawer-close"
          >
            <span
              class="ant-drawer-close-x"
            >
              <i
                class="anticon anticon-close"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="close"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </i>
            </span>
          </button>
          <div
            class="ant-drawer-body"
          >
            Here is content of Drawer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer have a title 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
      style="transform:translateX(100%);-ms-transform:translateX(100%);width:256px"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          class="ant-drawer-wrapper-body"
          style="overflow:auto;height:100%"
        >
          <div
            class="ant-drawer-header"
          >
            <div
              class="ant-drawer-title"
            >
              Test Title
            </div>
          </div>
          <button
            aria-label="Close"
            class="ant-drawer-close"
          >
            <span
              class="ant-drawer-close-x"
            >
              <i
                class="anticon anticon-close"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="close"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </i>
            </span>
          </button>
          <div
            class="ant-drawer-body"
          >
            Here is content of Drawer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer render correctly 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
      style="transform:translateX(100%);-ms-transform:translateX(100%);width:400px"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          class="ant-drawer-wrapper-body"
          style="overflow:auto;height:100%"
        >
          <button
            aria-label="Close"
            class="ant-drawer-close"
          >
            <span
              class="ant-drawer-close-x"
            >
              <i
                class="anticon anticon-close"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="close"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </i>
            </span>
          </button>
          <div
            class="ant-drawer-body"
          >
            Here is content of Drawer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer render top drawer 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-top"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
      style="transform:translateY(-100%);-ms-transform:translateY(-100%);height:400px"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          class="ant-drawer-wrapper-body"
        >
          <button
            aria-label="Close"
            class="ant-drawer-close"
          >
            <span
              class="ant-drawer-close-x"
            >
              <i
                class="anticon anticon-close"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="close"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                  />
                </svg>
              </i>
            </span>
          </button>
          <div
            class="ant-drawer-body"
          >
            Here is content of Drawer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
